tests/test_modeling_tf_common.py

```diff
@@ -2386,6 +2392,33 @@ def test_push_to_hub(self):
                 models_equal = False
         self.assertTrue(models_equal)
 
+    def test_push_to_hub_callback(self):
```

Comment:
This is just like the test above (including the same checks), but uses the callback mechanism to push the model.